### PR TITLE
Woloxapp conversation

### PR DESCRIPTION
**Pull request by ignaciosantise**
## Summary
* Implemented chat screen
* Added navigation bar in main and chat screen

## Pendings
* Adjust the screen when the keyboard shows up (iOS)
* Apply rounded style to profile images (iOS)

## Screenshots
screen shot 2017-04-04 at 10 33 34 am
screen shot 2017-04-04 at 10 33 47 am

### CODE REVIEW

File: src/components/ChatFooter/ChatFooter.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { connect } from 'react-redux';
+
+import styles from './ChatFooter.styles';
+import MessagingHandlers from '../../handlers/MessagingHandlers';
+
+const { number } = React.PropTypes;
+
+class ChatFooter extends Component {
+  state = {
+    inputValue: '',
```
Comment (mvbattan) [inline src/components/ChatFooter/ChatFooter.js]: I'd extract `''` to a variable

File: src/components/ChatFooter/ChatFooter.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    sendDisabled: true
+  };
+
+  handleInputChange = (text) => {
+    this.setState({
+      inputValue: text,
+      sendDisabled: (text === '')
```
Comment (mvbattan) [inline src/components/ChatFooter/ChatFooter.js]: I'd extract `''` to a variable

File: src/components/ChatFooter/ChatFooter.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    });
+  };
+
+  handleButtonPress = () => {
+    this.props.dispatch(MessagingHandlers.postMessage(
+      this.state.inputValue,
+      this.props.loggedUserId,
+      this.props.receiverId
+    ));
+    this.handleInputChange('');
```
Comment (mvbattan) [inline src/components/ChatFooter/ChatFooter.js]: I'd extract `''` to a variable

File: src/components/ChatFooter/ChatFooter.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  };
+
+  render() {
+    return (
+      <View style={styles.container}>
+        <TextInput multiline underlineColorAndroid="rgba(0,0,0,0)" style={styles.input} onChangeText={this.handleInputChange} value={this.state.inputValue} />
```
Comment (mvbattan) [inline src/components/ChatFooter/ChatFooter.js]: I'd use a line for every prop being passed to `TextInput`. As a rule of thumb, if there is two or more props being passed to a component, they should be split in multiple lines.

File: src/components/ChatFooter/ChatFooter.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        <Button title="send" onPress={this.handleButtonPress} disabled={this.state.sendDisabled} />
```
Comment (mvbattan) [inline src/components/ChatFooter/ChatFooter.js]: Same here

File: src/components/ChatFooter/ChatFooter.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+const mapStateToProps = (store) => {
+  return {
+    loggedUserId: store.users.loggedUserId
+  };
+};
+
+ChatFooter.propTypes = {
+  loggedUserId: number,
```
Comment (mvbattan) [inline src/components/ChatFooter/ChatFooter.js]: isRequired

File: src/components/ChatFooter/ChatFooter.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  receiverId: number
```
Comment (mvbattan) [inline src/components/ChatFooter/ChatFooter.js]: isRequired

File: src/components/ChatItem/ChatItem.js [hunk 8]
```diff
@@ -2,26 +2,28 @@ import React, { Component } from 'react';
...
+        contact
       }
     });
   };
 
   render() {
+    const profileStyle = isIos ? styles.profileImg_ios : styles.profileImg;
     return (
-      <TouchableHighlight underlayColor={lightGray} onPress={() => this.navigate(this.props.contact.id)}>
+      <TouchableHighlight underlayColor={lightGray} onPress={() => this.navigate(this.props.contact)}>
```
Comment (mvbattan) [inline src/components/ChatItem/ChatItem.js]: Same here

File: src/components/ChatMessage/ChatMessage.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+import Moment from 'moment';
+
+import styles from './ChatMessage.styles';
+
+const { shape, number, string } = React.PropTypes;
+
+class ChatMessage extends Component {
+  render() {
+    const viewStyle = (this.props.message.senderId === this.props.loggedUserId ? styles.ownerMessage : styles.notOwnerMessage);
+    const date = Moment(this.props.message.createdAt).format('hh:mm');
```
Comment (mvbattan) [inline src/components/ChatMessage/ChatMessage.js]: Well played

File: src/components/ChatView/ChatView.js [hunk 10]
```diff
@@ -1,23 +1,41 @@
...
     return (
-      <ListView
-        enableEmptySections
-        dataSource={this.ds.cloneWithRows(this.props.messages)}
-        renderRow={(rowItem) => <Text>{rowItem.body}</Text>}
-      />
+      <View style={styles.container}>
+        <ChatViewBar navigator={this.props.navigator} title={this.props.contact.username} />
+        <ListView
+          ref={(c) => { this.listview = c; }}
```
Comment (mvbattan) [inline src/components/ChatView/ChatView.js]: Variable names should be pronounceable.

File: src/components/ChatViewBar/ChatViewBar.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+const { instanceOf, string } = React.PropTypes;
+
+class ChatViewBar extends Component {
+  handleBackPressed = () => {
+    this.props.navigator.pop();
+  };
+
+  render() {
+    const profileStyle = isIos ? styles.profileImg_ios : styles.profileImg;
+    const containerStyle = isIos ? styles.container_ios : styles.container;
```
Comment (mvbattan) [inline src/components/ChatViewBar/ChatViewBar.js]: A better practice for doing this is to use just one `styles.profileImg` or `styles.container`, letting the `.styles.js` to solve this difference

File: src/components/ChatViewBar/ChatViewBar.styles.js [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+export default StyleSheet.create({
+  container: {
+    flexDirection: 'row',
+    height: 56,
+    backgroundColor: darkBlue,
+    alignItems: 'center',
+    justifyContent: 'space-between'
+  },
+  container_ios: {
```
Comment (mvbattan) [inline src/components/ChatViewBar/ChatViewBar.styles.js]: So, here, you can get rid of `container_ios` and remove duplicated code

File: src/components/TabView/TabView.js [hunk 13]
```diff
@@ -38,13 +41,20 @@ export default class TabView extends Component {
...
   render() {
     return (
-      <TabViewAnimated
-        style={styles.container}
-        navigationState={this.state}
-        renderScene={this._renderScene}
-        renderHeader={this._renderHeader}
-        onRequestChangeTab={this._handleChangeTab}
-      />
+      <View style={{flex: 1}}>
```
Comment (mvbattan) [inline src/components/TabView/TabView.js]: Avoid inline styles

File: src/components/ChatItem/ChatItem.js (hunk 8 above)
Comment (ignaciosantise) [inline src/components/ChatItem/ChatItem.js]: In this particular case, eslint is warning me not to separate props in different lines

File: src/components/TabViewBar/TabViewBar.js [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+
+  render() {
+    const containerStyle = isIos ? styles.container_ios : styles.container;
+    return (
+      <View style={containerStyle}>
+        <Text style={styles.title}>WoloxApp</Text>
+        <View style={styles.buttonContainer}>
+          <TouchableHighlight underlayColor="transparent" onPress={this.handleSearchClick}>
+            <Image style={styles.button} source={searchIcon} />
+          </TouchableHighlight>
```
Comment (mvbattan) [inline src/components/TabViewBar/TabViewBar.js]: I wonder if this `TouchableHighlight` can be extracted to another component

File: src/components/ChatItem/ChatItem.styles.js [hunk 15]
```diff
@@ -21,5 +18,10 @@ export default StyleSheet.create({
     height: profileDim,
     borderRadius: 100,
     margin: 5
+  },
+  profileImg_ios: {
```
Comment (mvbattan) [inline src/components/ChatItem/ChatItem.styles.js]: Have you tried to reuse `profileImg` with `borderRadius: 0` ?

File: src/components/ChatViewBar/ChatViewBar.js [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+  handleBackPressed = () => {
+    this.props.navigator.pop();
+  };
+
+  render() {
+    return (
+      <View style={styles.container}>
+        <TouchableHighlight underlayColor="transparent" onPress={this.handleBackPressed}>
+          <Image style={styles.back} source={backImg} />
+        </TouchableHighlight>
```
Comment (mvbattan) [inline src/components/ChatViewBar/ChatViewBar.js]: This `TouchableHighlight` is a way similar to a previous one. Mmmm, for now, I'd let this as is, but keep in mind that if that there is a pattern being repeated cross components, this pattern might be a new component.

File: src/components/ChatFooter/ChatFooter.js [hunk 17]
```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import { TextInput, View, Button } from 'react-native';
+import { connect } from 'react-redux';
+
+import styles from './ChatFooter.styles';
+import MessagingHandlers from '../../handlers/MessagingHandlers';
+
+const { number } = React.PropTypes;
```
Comment (pabloferro) [inline src/components/ChatFooter/ChatFooter.js]: Just a comment, I think it's clearer if you use React.PropTypes.number directly in proptypes.
You don't need to change it here.

Comment (ignaciosantise) [inline src/components/ChatFooter/ChatFooter.js]: ok!

**Pull request merged**